packages/react-devtools-extensions/src/main.js

```diff
@@ -25,6 +25,25 @@ const LOCAL_STORAGE_SUPPORTS_PROFILING_KEY =
 
 const isChrome = getBrowserName() === 'Chrome';
 
+const cachedNetworkEvents = new Map();
+
+// Cache JavaScript resources as the page loads them.
+// This helps avoid unnecessary duplicate requests when hook names are parsed.
+// Responses with a Vary: 'Origin' migt not match future requests.
+// This lets us avoid a possible (expensive) cache miss.
+// For more info see: github.com/facebook/react/pull/22198
+chrome.devtools.network.onRequestFinished.addListener(
+  function onRequestFinished(event) {
+    const {method, url} = event.request;
+    if (method === 'GET' && url.indexOf('.js') > 0) {
```

Comment:
i was wondering, would this also happen to record source map files that are downloaded by chrome dev tools? i.e. should we also check for files with `.map` and cache those as well?

Reply:
if so, maybe this could be something to add in a follow up... might be worth landing and iterating

Reply:
> i was wondering, would this also happen to record source map files that are downloaded by chrome dev tools? i.e. should we also check for files with `.map` and cache those as well?

Great question!

It _would_ cache those as well, since the `.js` check would match the source maps' `.js.map` extension, but I'm not sure this API would even report those– since the page wouldn't be the thing that loaded them, rather, the browser's DevTools itself.

But we don't use the `fetchFileWithCaching` helper function for source-maps because it's unlikely they will have been loaded to begin with (and I wanted to avoid the extra overhead of serializing the huge source map text through `postMessage` since it's likely to be a cache miss anyway).

Reply:
yeah, was basically wondering if the API would report them in the first place. i wonder if we can hook into this api on the chrome dev tools context and then cache source map files separately? but that's probably a stretch. might be interesting to double check if they actually get reported though.

> It would cache those as well, since the .js check would match the source maps' .js.map extension

should we be more strict here about only caching javascript files?

Reply:
> should we be more strict here about only caching javascript files?

I think we are pretty strict, since we check `mimeType` too:

```js
if (method === 'GET' && url.indexOf('.js') > 0) {
  const {mimeType} = event.response.content;
  if (mimeType === 'application/x-javascript') {
    cachedNetworkEvents.set(url, event);
  }
}
```

TBH now that I think of it, source-maps _wouldn't_ be cached here since their mime type is likely `application/json` or something. 😄